pygmt/clib/conversion.py

```diff
@@ -101,14 +101,19 @@ def dataarray_to_matrix(grid):
                     dim
                 )
             )
-        region.extend([coord.min(), coord.max()])
+        region.extend(
+            [
+                coord.min() - coord_inc / 2 * grid.gmt.registration,
+                coord.max() + coord_inc / 2 * grid.gmt.registration,
+            ]
+        )
         inc.append(coord_inc)
 
-    if any([i < 0 for i in inc]):  # Sort grid when there are negative increments
+    if any(i < 0 for i in inc):  # Sort grid when there are negative increments
         inc = [abs(i) for i in inc]
         grid = grid.sortby(variables=list(grid.dims), ascending=True)
 
-    matrix = as_c_contiguous(grid.values[::-1])
+    matrix = as_c_contiguous(grid[::-1].values)
```

Comment:
It seems like we practically **always** create a memory copy here because the numpy view on a flipped grid `grid[::-1]` is never (or hardly ever?) c_contiguous. I don't know if there's a way around this, but if we need to switch to use `GMT_IS_DUPLICATE` for stability in #517, wouldn't that mean we create 2 memory copies?!

Reply:
I'll try using [`xarray.DataArray.copy`](xarray.pydata.org/en/v0.16.0/generated/xarray.DataArray.copy.html) at 5a685b2fcf36768fccc345c92e4fab2afc19dbe3 to force a deep copy (that should return a c_contiguous array) before it enters the `as_c_contiguous` function (which may or may not return a numpy copy). It seems to give a more consistent black image, rather than the reddish-brown one :sweat_smile:

Reply:
Reverted 5a685b2fcf36768fccc345c92e4fab2afc19dbe3 at 60c791de17382495c181bddc5959282a7c25db7d, will stick to just `grid[::-1].values` for now (i.e. flipping the grid left right in xarray rather than in numpy).